frontend/src/redux/museumArtworkActions.js

```diff
@@ -0,0 +1,88 @@
+const getIIIFLevel = (artwork, displayWidth) => {
+  return {
+    url: 'https://www.artic.edu/iiif/2/' + artwork.image_id + '/full/' + displayWidth + ',/0/default.jpg',
+    width: displayWidth,
+    height: Math.floor(artwork.thumbnail.height *
+      displayWidth / artwork.thumbnail.width),
+  };
+};
+
+const getArtworkInfo = (artwork) => {
+  const alt = artwork.thumbnail.alt_text;
+  const linkToImage = getIIIFLevel(artwork, 500);
+  const title = artwork.title;
+  const artworkInfo = new Map();
+  artworkInfo.set('alt', alt);
+  artworkInfo.set('url', linkToImage.url);
+  artworkInfo.set('title', title);
+  return artworkInfo;
+};
+
+function getMuseumArtworks(param) {
+  const API = 'https://aggregator-data.artic.edu/api/v1/'+param;
```

Comment:
Building URLs by hand can easily go wrong, so it's more robust to use the builder pattern so that the JS engine always gets it right.

In this case, I'd pass in the relative path and a map of params, and then do something like this:

```suggestion
function getMuseumArtworks(path, params) {
  const apiUrl = 'https://aggregator-data.artic.edu/api/v1/';
  apiUrl.pathname += path;
  if (params) {
    for (let [key, value] of params) {
      apiUrl.searchParams.append(key, value);
    }
    const API = apiUrl.toString();
  }
```

Reply:
Thank you for this feedback!